### PR TITLE
Change from flake8-putty to per-file-ignores

### PR DESCRIPTION
**Pull request by samuelhwilliams**
## Summary
Add flake8 per-file-ignores, in place of flake8-putty, so that we can support newer Python3 syntax like eg f-strings and type annotations.

Flake8 report:
```
/Users/samuelwilliams/git/digitalmarketplace-buyer-frontend/venv/bin/flake8 .
./app/main/views/__init__.py:0:1: X100 Superfluous per-file-ignores for C901
./app/main/views/crown_hosting.py:0:1: X100 Superfluous per-file-ignores for C901
./app/main/views/digital_services_framework.py:0:1: X100 Superfluous per-file-ignores for C901
./app/main/views/feedback.py:0:1: X100 Superfluous per-file-ignores for C901
./app/main/views/g_cloud.py:0:1: X100 Superfluous per-file-ignores for C901
./app/main/views/login.py:0:1: X100 Superfluous per-file-ignores for C901
./app/main/views/marketplace.py:0:1: X100 Superfluous per-file-ignores for C901
./app/main/views/suppliers.py:0:1: X100 Superfluous per-file-ignores for C901
./tests/main/helpers/test_search_helpers.py:185:13: F601 dictionary key 'page' repeated with different values
./tests/main/helpers/test_search_helpers.py:188:13: F601 dictionary key 'page' repeated with different values
make: *** [test-flake8] Error 1
```

### CODE REVIEW

File: tests/main/helpers/test_search_helpers.py [hunk 1]
```diff
@@ -182,7 +182,6 @@ def test_clean_request_args(self):
             'question1': 'true',
             'question2': 'true',
             'question3': ['option1', 'option2'],
-            'page': 'false',
```
Comment (katstevens) [inline tests/main/helpers/test_search_helpers.py]: Is this a flake8 related change?

Comment (samuelhwilliams) [reply tests/main/helpers/test_search_helpers.py]: It is indeed. Apologies - updated the PR with the flake8 report after changing to the new extension. It reported this: `./tests/main/helpers/test_search_helpers.py:185:13: F601 dictionary key 'page' repeated with different values`

Comment (katstevens) [reply tests/main/helpers/test_search_helpers.py]: Aha! Cool, thanks for adding the report.

**Comment by Wynndow** [comment]
Are we losing anything by dropping flake8-putty?

**Comment by samuelhwilliams** [comment]
I don't believe so - we only use it to give us the ability to ignore flake8 errors on a per-file basis.

**Comment by Wynndow** [comment]
In which case have a 👍
